src/frontends/tensorflow_common/src/op/select.cpp

```diff
@@ -25,6 +29,26 @@ OutputVector translate_select_base_op(const NodeContext& node,
                                       const Output<Node>& condition,
                                       const Output<Node>& x,
                                       const Output<Node>& y) {
+    auto complex_type_mark_x = as_type_ptr<ComplexTypeMark>(x.get_node_shared_ptr());
+    auto complex_type_mark_y = as_type_ptr<ComplexTypeMark>(y.get_node_shared_ptr());
+
+    if (complex_type_mark_x && complex_type_mark_y) {
+        auto const_1 = make_shared<v0::Constant>(element::i32, Shape{1}, 1);
+        auto cond_shape = make_shared<v3::ShapeOf>(condition, element::i32);
+        auto new_cond_shape = make_shared<v0::Concat>(OutputVector{const_1, cond_shape}, 0);
```

Comment:
I think it should be `auto new_cond_shape = make_shared<v0::Concat>(OutputVector{cond_shape, const_1}, 0);`
since auxiliary dim is the last one and we need to provide broadcasting along that dimension as well.